src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,414 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Binance.com REST plugin links:
+# REST API: https://docs.kraken.com/rest/
+# Authentication: https://docs.kraken.com/rest/#section/Authentication
+# Endpoint: https://api.kraken.com
+
+# CCXT documentation:
+# https://docs.ccxt.com/en/latest/index.html
+
+import pytz
+import logging
+from datetime import datetime
+from typing import List, Optional
+
+from ccxt import Exchange, kraken
+
+from dali.abstract_ccxt_input_plugin import (
+    AbstractCcxtInputPlugin,
+)
+from dali.ccxt_pagination import (
+    AbstractPaginationDetailSet,
+)
+from rp2.logger import create_logger
+from dali.abstract_transaction import AbstractTransaction
+from dali.in_transaction import InTransaction
+from dali.intra_transaction import IntraTransaction
+from dali.out_transaction import OutTransaction
+from dali.configuration import Keyword, _FIAT_SET
+
+
+# keywords
+_OUT: str = "out"
+_IN: str = "in"
+_INTRA: str = 'intra'
+_RESULT: str = 'result'
+_COUNT: str = 'count'
+_LEDGER: str = 'ledger'
+_TRADES: str = 'trades'
+_OFFSET: str = 'ofs'
+_TYPE: str = 'type'
+_REFID: str = 'refid'
+_TIMESTAMP: str = 'time'
+_FEE: str = 'fee'
+_ASSET: str = 'asset'
+_BASE: str = 'base'
+_BASE_ID: str = 'baseId'
+_AMOUNT: str = 'amount'
+_COST: str = 'cost'
+_PRICE: str = 'price'
+_WITHDRAWAL: str = 'withdrawal'
+_DEPOSIT: str = 'deposit'
+_MARGIN: str = 'margin'
+_TRADE: str = 'trade'
+_ROLLOVER: str = 'rollover'
+_TRANSFER: str = 'transfer'
+_SETTLED: str = 'settled'
+_CREDIT: str = 'credit'
+_STAKING: str = 'staking'
+_SALE: str = 'sale'
+
+# Record Limits
+_TRADE_RECORD_LIMIT: int = 50
+
+
+class InputPlugin(AbstractCcxtInputPlugin):
+
+    __EXCHANGE_NAME: str = "kraken"
+    __PLUGIN_NAME: str = "kraken_REST"
+    __DEFAULT_THREAD_COUNT: int = 1
+    __CACHE_FILE: str = 'kraken.pickle'
+
+    def __init__(
+        self,
+        account_holder: str,
+        api_key: str,
+        api_secret: str,
+        native_fiat: str,
+        username: Optional[str] = None,
+        thread_count: Optional[int] = __DEFAULT_THREAD_COUNT,
+        use_cache: Optional[bool] = True,
+    ) -> None:
+        self.__api_key = api_key
+        self.__api_secret = api_secret
+
+        # We will have a default start time of July 27th, 2011 since Kraken Exchange officially launched on July 28th.
+        super().__init__(account_holder, datetime(2011, 7, 27, 0, 0, 0, 0), native_fiat, thread_count)
+        self.__username = username
+        self.__logger: logging.Logger = create_logger(f"{self.__EXCHANGE_NAME}")
+        self.__timezone = pytz.timezone('UTC')
+        self._initialize_client()
+        self._client.load_markets()
+        self.baseId_to_base = {value[_BASE_ID]: value[_BASE] for key, value in self._client.markets_by_id.items()}
```

Comment:
All identifiers are lowercase with underscores, so it would be `base_id_to_base `